Shared/Supporting Files/Extensions/Array+RawRepresentable.swift

```diff
@@ -15,7 +15,7 @@
 import Foundation
 
 /// An extension allowing an array to be used with the app storage property wrapper.
-extension Array: RawRepresentable where Element == String {
+extension Array: @retroactive RawRepresentable where Element == String {
```

Comment:
We really shouldn't be doing this. Could the adoption of the protocol itself be removed?

```swift
extension Array<String> {
    ...
}
```

Reply:
This is used for #313 `FavoritesView`'s `AppStorage` to serialized an array of strings. If we don't add this extension, then we might need to roll our own UserDefault managing types instead. (which I tried by then and is possible, but involves manually insert/remove from user defaults so it is less clean than this solution)

Reply:
I'm not suggesting getting rid of the functionality. It could still be used by changing the app storage type to `String` and then adding a computed property:

```swift
@AppStorage(.favoriteSampleNames) private var favoriteNamesString: String = ""

private var favoriteNames: [String] {
    get { .init(rawValue: favoriteNamesString) }
    nonmutating set { favoriteNamesString = newValue.rawValue }
}
```

I also suggest renaming `rawValue` to `jsonString`.

Reply:
Thanks for the new perspective 👍  Haven't thought we could just store a string instead of an array.